core/templates/dev/head/domain/dashboard/DashboardBackendApiService.js

```diff
@@ -33,7 +33,26 @@ oppia.factory('DashboardBackendApiService', [
     });
   };
 
+  var _fetchExplorationStats = function(explorationId) {
+    var requestUrl = (
+      '/dashboardhandler/explorationstats/?exp_id=' + explorationId);
+    return $q(function(resolve, reject) {
+      $http.get(requestUrl, {
+        cache: true
+      }).then(function(response) {
+        if (resolve) {
```

Comment:
On further inspection, I think constructing a promise in the first place is entirely unnecessary here. `http.get` already returns a promise, and promises are composable, so you shouldn't need to wrap around it with another promise.